src/state/shellscripts/sandbox_exec.sh

```diff
@@ -13,12 +13,29 @@ add_mounts() {
 
 add_mounts rw "${TMPDIR:-/tmp}"
 
+# C compilers using `ccache` will write to a shared cache directory
+# that remain writeable. ccache seems widespread in some Fedora systems.
+add_ccache_mount() {
+  if command -v ccache > /dev/null; then
+      CCACHE_DIR=$HOME/.ccache
+      ccache_dir_regex='cache_dir = (.*)$'
+      local IFS=$'\n'
+      for f in $(ccache --print-config); do
+        if [[ $f =~ $ccache_dir_regex ]]; then
+          CCACHE_DIR=${BASH_REMATCH[1]}
+        fi
```

Comment:
The `bwrap` script is bash-only, but this one is declared as `sh`; is this still OK ?

Reply:
Oh, I hadn't thought of that. If you are not willing to depend on `bash` on OSX, maybe we could just give up on `ccache` support in OSX for the time being? I'm not very excited at trying to figure out the right grep+sed magic -- although I suppose that `ccache --print-config | grep -o 'cache_dir = .*$' | sed 's/cache_dir //g'` should work, I cannot test it.

Reply:
Maybe that's the safest, let's ask an OSX user (@avsm ?) about it ?

Reply:
I'm a little confused: bwrap is Linux-only, where you can (more-or-less) depend on bash being available.  The bwrap script should never run on OSX should it?

Either way, bash is the default on osx, but an ancient version (3.2.57), so you need to be careful not to use bash4-isms. The portability danger is on OpenBSD, where `/bin/sh` is `ksh` and not `bash` or `dash`.

Reply:
it's actually the `sandbox_exec` script that we are discussing here (and which was not bash-only until now) ; but I managed to make the `bwrap` and `sandbox_exec` scripts quite similar, at least for their interesting parts.

@gasche: OK to merge then, just make the `#!` bash for `sandbox_exec` and we're good. Thanks!